packages/circuit-ui/components/Autocomplete/components/AutocompleteResults/AutocompleteResults.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2025, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import { forwardRef, type ReactNode, useRef } from 'react';
+
+import { utilClasses } from '../../../../styles/utility.js';
+import {
+  SuggestionBox,
+  type SuggestionBoxProps,
+} from '../SuggestionBox/SuggestionBox.js';
+import { Hr } from '../../../Hr/index.js';
+import { clsx } from '../../../../styles/clsx.js';
+import { Spinner } from '../../../Spinner/index.js';
+import { Button, type ButtonProps } from '../../../Button/index.js';
+
+import classes from './AutocompleteResults.module.css';
+
+export type AutocompleteResultsProps = SuggestionBoxProps & {
+  /**
+   * Custom content to display while loading suggestions.
+   */
+  loadingLabel?: string;
+  /**
+   * Custom content to display when no suggestions are available.
+   */
+  noResultsMessage?: ReactNode;
+  /**
+   * An optional function that allows to add more items to the bottom the suggestion list currently displayed.
+   * If this function is provided, a "Load more" button will be displayed at the bottom of the suggestion list.
+   * Use this to implement lazy loading of suggestions.
+   */
+  loadMore?: () => void;
+  /**
+   * An optional button to display below the Autocomplete suggestions.
+   */
+  action?: Omit<ButtonProps, 'variant' | 'size'>;
+  resultsSummary: string;
+  isModal?: boolean;
+};
+
+export const AutocompleteResults = forwardRef<
+  HTMLDivElement,
+  AutocompleteResultsProps
+>(
+  (
+    {
+      isLoading,
+      isLoadingMore,
+      loadingLabel,
+      noResultsMessage,
+      suggestions,
+      value,
+      onSuggestionClicked,
+      label,
+      activeSuggestion,
+      loadMore,
+      action,
+      suggestionIdPrefix,
+      allowNewItems,
+      searchText,
+      resultsSummary,
+      isModal,
+      loadMoreLabel,
+    },
+    ref,
+  ) => {
+    const actionsRef = useRef<HTMLDivElement>(null);
+    return (
+      <div
+        ref={ref}
+        className={clsx(!isModal && classes.modal)}
+        style={{
+          marginBottom: isModal
+            ? `${actionsRef?.current?.getBoundingClientRect().height}px`
```

Comment:
Measuring DOM elements during render is brittle (the element might not have been rendered yet) and costly (DOM measurements are relatively expensive calculations). This should either happen in a [`useLayoutEffect`](https://react.dev/reference/react/useLayoutEffect), or better not at all. The height is a static value (button height + plus padding), isn't it?